### PR TITLE
webui: add option to report an issue to global menu

### PR DESCRIPTION
**Pull request by rvykydal**
Addresses https://issues.redhat.com/browse/INSTALLER-3610.

This first stab PR needs:
- [x] design / microcopy review and update
- [x] tests

### CODE REVIEW

**Comment by rvykydal** [comment]
Screenshot from 2023-08-15 09-34-31
Screenshot from 2023-08-15 09-34-56
Screenshot from 2023-08-15 09-37-16
Screenshot from 2023-08-15 09-37-33

File: ui/webui/src/components/HeaderKebab.jsx [hunk 1]
```diff
@@ -152,6 +162,11 @@ export const HeaderKebab = () => {
               isModalOpen={isAboutModalOpen}
               setIsAboutModalOpen={setIsAboutModalOpen}
             />}
+            {isReportIssueOpen &&
+            <UserIssue
+              reportLinkURL={reportLinkURL}
+              setIsReportIssueOpen={setIsReportIssueOpen}
+            />}
```
Comment (VladimirSlavik) [inline ui/webui/src/components/HeaderKebab.jsx:169]: Probably fine for now, but I wonder if these `isSomethingOpen` variables are usable at scale? We'll eventually have many of these modals openable from the menu, right? Or am I reading this wrong and it's certain there won't be code doing `if (isAaOpen || isBbOpen || isCcOpen) ...`?

**Comment by rvykydal** [comment]
> Addresses https://issues.redhat.com/browse/INSTALLER-3610.
> 
> This first stab PR needs:
> 
>     * [ ]  design / microcopy review and update

@garrett can you please take look ?

File: ui/webui/src/components/HeaderKebab.jsx (hunk 1 above)
Comment (VladimirSlavik) [reply ui/webui/src/components/HeaderKebab.jsx:168]: Yes, I agree it's fine now, especially as I didn't find the spaghetti condition anywhere. That's also why I asked - I thought perhaps I'm missing something and the spaghetti can't happen, or is resolved automagically by the framework instead. (Which would be fine too!)

**Comment by rvykydal** [comment]
Added the tests.

**Comment by garrett** [comment]
Hey! Is this PR just about the menu or other things?

(There's a strange alert that doesn't make any sense about "additional disks". But that's not part of this PR.)

If it's just about adding the error reporting as a menu item, then it's fine. If it's about the text in the modal dialog, then it's weird and would need to be adjusted, including both the heading and the description text.

**Comment by rvykydal** [comment]
> Hey! Is this PR just about the menu or other things?
> 
> (There's a strange alert that doesn't make any sense about "additional disks". But that's not part of this PR.)
> 
> If it's just about adding the error reporting as a menu item, then it's fine. If it's about the text in the modal dialog, then it's weird and would need to be adjusted, including both the heading and the description text.

Yes, I am mainly asking for help with that :)
The purpose is to allow user to file an issue/bug at any point of the installation.
I think the tricky part is that the description of the issue should be filed by the user later - on the bugzilla enter bug page.

> 
> Additionally, the log size (and therefore modal) is way, way, way too small if we expect someone to read through the log and possibly make changes. (Refer to the mockup.)

Yes, I will fix it (also for Critical Error reports).

**Comment by garrett** [comment]
>  I think the tricky part is that the description of the issue should be filed by the user later - on the bugzilla enter bug page.

Yeah, this is really sub-optimal. We should be collecting all that info in Anaconda itself. I guess that's for F40 and beyond, right?

This is nearly the same as the critical issue reporting modal, except there's no icon, the title changed, there's some description, and no error message. I've made an offline variant of this one too (which is the same as the critical offline one, where the button is disabled and there's an error message with the DisconnectedIcon PF icon (also known as pficon-disconnected) — until the network status changes to online).

Optional bug report_ simple, details, larger

Optional bug report_ simple, details, larger, no network

Just like the critical issue reporting, your screenshot indicates that it's too small. This needs to be a very tall dialog to have a very tall log view, if we expect people to view it and (perhaps) make changes.

**Review by KKoukiou — APPROVED** [review]

**Comment by rvykydal** [comment]
Updated for @garrett's review:

**Comment by rvykydal** [comment]
> ... I've made an offline variant of this one too (which is the same as the critical offline one, where the button is disabled and there's an error message with the DisconnectedIcon PF icon (also known as pficon-disconnected) — until the network status changes to online).

That is great point, I'd address it for both dialogs as a separate PR?

**Review by garrett — APPROVED** [review]
With the changes, it looks good! Thanks!

> That is a great point, I'd address it for both dialogs as a separate PR?

Yes, please do add "offline support" for both in a separate PR, so we can merge this one in and fix up what happens when there's no network connectivity for both. Thanks!

**Comment by rvykydal** [comment]
Rebased on top of pf 5 changes.

**Pull request merged**